Sync v2-9-stable with v2-9-test to release 2.9.0

ephraimbuddy opened the pull request:
Time for `2.9.0b2`!

ephraimbuddy committed:
Update version to 2.9.0

ephraimbuddy committed:
Update RELEASE_NOTES.rst

potiuk committed:
Build constraints instead of pulling them for non-main builds

PROD image - when built in non-main branch - needs constraints
generated for PyPI builds. Those constraints are generated in
"generate-constraints" build but we should not add a dependency
for that job, because it will slow-down the regular main builds and
PRs. Instead, we should generate the constraints again - this is now
rather quick thanks to `uv` so it should not introduce much delay in
the non-main builds.

(cherry picked from commit 1d3f92b3a7ca81a08bcbf1def421b094b1b50898)

potiuk committed:
Fix epoch time for reproducible build

potiuk committed:
Fix failure of additional prod image builds in non-main build

As of #38533 - we are building locally constraints for non-main
production image builds. This change required CI images to be used
and pulled before they were used. For "additional" image checks
image-tag used in this case contained the "extra" prefix and the
CI image could not be pulled.

This change fixes this, it also displays the right Python version
when pulling CI image and forces it's use in the pull/constraint
steps to be explicitly used when provider packages are built.

(cherry picked from commit 18c2e94a25e0f668329a9291f8615b6455ad43e9)

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,285 @@
+
+The Audit Log ``event`` name for REST API events will be prepended with ``api.`` or ``ui.``, depending on if it came from the Airflow UI or externally.
+
+Airflow 2.9.0 is the first release that officially supports Python 3.12 (#38025)
```
```suggestion
Official support for Python 3.12 (#38025)
```
We can shorten a bit :)

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,285 @@
+- Support multiple XCom output in the BaseOperator (#37297)
+- AIP-58: Add object storage backend for xcom (#37058)
+- Introduce ``DatasetOrTimeSchedule`` (#36710)
+- Add ``on_skipped_callback`` in to ``BaseOperator`` (#36374)
```
```suggestion
- Add ``on_skipped_callback`` to ``BaseOperator`` (#36374)
```

eladkal commented on RELEASE_NOTES.rst:
This is provider only - we can skip it from airflow core change log

eladkal commented on README.md:
```diff
@@ -98,9 +98,9 @@ Airflow is not a streaming solution, but it is often used to process real-time d
-| Python      | 3.8, 3.9, 3.10, 3.11, 3.12 | 3.8, 3.9, 3.10, 3.11        |
+| Python      | 3.8, 3.9, 3.10, 3.11, 3.12 | 3.8, 3.9, 3.10, 3.11, 3.12  |
 | Platform    | AMD64/ARM64(\*)            | AMD64/ARM64(\*)             |
 | Kubernetes  | 1.26, 1.27, 1.28, 1.29     | 1.25, 1.26, 1.27, 1.28, 1.29|
```
i think we no longer support K8s 1.25 in Airflow 2.9?

ephraimbuddy commented on RELEASE_NOTES.rst:
```suggestion
```

potiuk committed:
Fix to long constraint generation in `pip` builds (#38541)

Even if we are using `pip` to generate image, we still want to make sure
constraints generation is done by UV as it is far more efficient.

This PR fixes too long constraint generation we seein v2-* branch now.

It also adds missing break in constraint generation that produces
a misleading warning that chicken-egg provider package has not been
used.

(cherry picked from commit 8bf20dce1a0f2c1275e240ae33d5e5d614fa6fe8)

potiuk committed:
Optimize preparing packages for PROD image builds (#38539)

With this change provider and airflow packages are prepared only
once for all PROD image builds and uploaded as artifacts. This saves
~ 4 minutes per PROD image build (which is multiplied by the number
of Python versions we have and by all the additional PROD image
checks we have.

(cherry picked from commit 6ebbda515eee4b9739997290132eb62d775907fe)
